src/modules/roc_pipeline/receiver_session_group.cpp

```diff
@@ -108,6 +114,16 @@ void ReceiverSessionGroup::create_session_(const packet::PacketPtr& packet) {
         return;
     }
 
+    packet::PacketPtr stored_packet, stored_packet_next;
+    for (stored_packet = packet_ring_buffer_.front(); stored_packet;
+         stored_packet = stored_packet_next) {
+        if (!sess->handle(stored_packet)) {
+            break;
```

Comment:
Could you explain why do we breaking the loop here?

Reply:
I was thinking if the session cannot handle the current packet, then it won't be able to handle the remaining packets.

Reply:
I don't see why? handle() will return false if the packet is not intended for this session. If the first packet in the ring buffer is for another session, it doesn't mean that the second, or any other packet is not for this session, right?

Reply:
I see 👍